compileoptions.cmake

```diff
@@ -36,7 +36,8 @@ if (CLR_CMAKE_PLATFORM_UNIX)
   # There are constants of type BOOL used in a condition. But BOOL is defined as int
   # and so the compiler thinks that there is a mistake.
   add_compile_options(-Wno-constant-logical-operand)
-
+  add_compile_options(-Wno-null-dereference)
```

Comment:
I am not happy about adding new warning disabling. I have spent quite some time by making sure the disabled warning set it the minimum necessary. We should rather fix the code that's causing them. However, building the most recent master branch on amd64 with clang 3.9.1, I am not getting any of these warnings. So for some reason, it looks like the warnings are reported for arm64 builds. Could you please share the build error logs?